perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java

```diff
@@ -128,6 +134,27 @@ public Member createSpecificMember(PerunSession sess, Vo vo, Candidate candidate
 		return getMembersManagerBl().createSpecificMember(sess, vo, candidate, specificUserOwners, specificUserType, groups);
 	}
 
+	public Member createSponsoredAccount(PerunSession sess, Map<String, String> params, String namespace, ExtSource extSource, String extSourcePostfix, Vo vo, int loa) throws InternalErrorException, PrivilegeException, UserNotExistsException, ExtSourceNotExistsException, UserExtSourceNotExistsException, WrongReferenceAttributeValueException, LoginNotExistsException, PasswordCreationFailedException, ExtendMembershipException, AlreadyMemberException, GroupOperationsException, PasswordStrengthFailedException, PasswordOperationTimeoutException, WrongAttributeValueException {
+		Utils.checkPerunSession(sess);
+		Utils.notNull(extSource, "extSource");
+		Utils.notNull(namespace, "namespace");
+		Utils.notNull(vo, "vo");
+		Utils.notNull(extSourcePostfix, "extSourcePostfix");
+
+		if (!AuthzResolver.isAuthorized(sess, Role.REGISTRAR, vo)) {
```

Comment:
Registrar role is not associated with VO (as far as I know). Generally REGISTRAR can do anything regarding user registration and password reservation/change. I would prefer using VOADMIN role, but if I remember, @licehammer wanted to use REGISTRAR. So please just remove ``vo`` param from call ``isAuthorized``.